tests/testthat/test-helpers_knitr_Stopping.R

```diff
@@ -0,0 +1,413 @@
+      "smaller of the doses given by the end-of-trial TD estimate and the G* ",
+      "estimate is less than or equal to 4."
+    )
+  )
+})
+
+test_that("knit_print.StoppingTDCIRatio works correctly", {
+  expect_equal(
+    knit_print(
+      StoppingTDCIRatio(target_ratio = 4, prob_target = 0.3, report_label = "TDCI_RATIO"),
+      asis = FALSE
+    ),
+    paste0(
+      "TDCI_RATIO: If, at the next best dose, the ratio of the upper and lower ",
+      "limits of the posterior 95% credible interval for the probability of ",
+      "toxicity is less than or equal to 4."
+    )
+  )
+  expect_equal(
+    knit_print(
+      StoppingTDCIRatio(target_ratio = 4, prob_target = 0.3),
+      asis = FALSE,
+      dose_label = "the MTD"
+    ),
+    paste0(
+      "If, at the MTD, the ratio of the upper and lower ",
+      "limits of the posterior 95% credible interval for the probability of ",
+      "toxicity is less than or equal to 4."
+    )
+  )
+})
+
+test_that("knit_print.StoppingTargetBiomarker works correctly", {
+  expect_equal(
+    knit_print(
+      StoppingTargetBiomarker(target = c(0.9, 1.0), prob = 0.5, report_label = "BIOMARKER"),
+      asis = FALSE
+    ),
+    paste0(
+      "BIOMARKER: If, at the next best dose, the posterior probability that the ",
+      "target biomarker is in the range (0.90, 1.00), relative to the maximum ",
+      "value of the target biomarker, is 50% or more."
+    )
+  )
+  expect_equal(
+    knit_print(
+      StoppingTargetBiomarker(target = c(300, 500), is_relative = FALSE, prob = 0.75),
+      asis = FALSE,
+      dose_label = "the MTD",
+      biomarker_label = "hs-CRP"
+    ),
+    paste0(
+      "If, at the MTD, the posterior probability that hs-CRP ",
+      "is in the range (300.00, 500.00) is 75% or more."
+    )
+  )
+})
+
+test_that("knit_print.StoppingLowestDoseHSRBeta works correctly", {
+  expect_equal(
+    knit_print(
+      StoppingLowestDoseHSRBeta(target = 0.33, a = 1, b = 1, report_label = "HSR"),
+      asis = FALSE
+    ),
+    paste0(
+      "HSR: If, using a Hard Stopping Rule with a prior of Beta(1, 1), the ",
+      "lowest dose in the dose grid has a posterior probability of toxicity of ",
+      "95% or more."
+    )
+  )
+  expect_equal(
+    knit_print(
+      StoppingLowestDoseHSRBeta(target = 0.33, a = 3, b = 2, prob = 0.5),
+      asis = FALSE
+    ),
+    paste0(
+      "If, using a Hard Stopping Rule with a prior of Beta(3, 2), the ",
+      "lowest dose in the dose grid has a posterior probability of toxicity of ",
+      "50% or more."
+    )
+  )
+})
+
+test_that("knit_print.StoppingMTDCV works correctly", {
+  expect_equal(
+    knit_print(
+      StoppingMTDCV(target = 0.33, thresh = 0.5, report_label = "MTD_CV"),
+      asis = FALSE
+    ),
+    paste0(
+      "MTD_CV: If the posterior estimate of the robust coefficient of variation ",
+      "of the MTD, expressed as a percentage, is both greater than zero and less ",
+      "than or equal to 50%."
```

Comment:
```suggestion
      "MTD_CV: If the posterior estimate of the robust coefficient of variation ",
      "of the MTD (targeting 33%), is below 50%."
```
(mentioning the `target` of the MTD, and simplifying the condition - the MTD_CV will always be non-negative)

Reply:
OK.  I thought the CV should always be > 0.  The code does test for CV > 0, though.  
```
do_stop <- (mtd_cv <= stopping@thresh_cv) && (mtd_cv >= 0)
```
Only a small thing, but I can remove if you wish.

Reply:
yeah can remove there